offline/cable_cru_TRENDY.F90

```diff
@@ -1,1318 +1,1101 @@
+               vwindNo, fdiffNo
+
+  ! Set up and read the namelist
+  NAMELIST /MetNames/ rainNames, lwdnNames, swdnNames, presNames, qairNames,&
+                    TmaxNames, TminNames, uwindNames, vwindNames, fdiffNames,&
+                    rainNo, lwdnNo, swdnNo, presNo, qairNo, TmaxNo, TminNo,&
+                    uwindNo, vwindNo, fdiffNo
+
+  CALL get_unit(nmlUnit)
+  OPEN(nmlUnit, FILE = "met_names.nml", STATUS = 'old', ACTION = 'read')
+  READ(nmlUnit, NML = MetNames)
+  CLOSE(nmlUnit)
+
+  ! Allocate memory for the names in the SpatioTemporalDataset struct
+  ALLOCATE(STDatasets(1)%VarNames(rainNo), STDatasets(2)%VarNames(lwdnNo),&
+    STDatasets(3)%VarNames(swdnNo), STDatasets(4)%VarNames(presNo),&
+    STDatasets(5)%VarNames(qairNo), STDatasets(6)%VarNames(TmaxNo),&
+    STDatasets(7)%VarNames(TminNo), STDatasets(8)%VarNames(uwindNo),&
+    STDatasets(9)%VarNames(vwindNo), STDatasets(10)%VarNames(fdiffNo))
+
+  STDatasets(1)%VarNames = rainNames(1:rainNo)
+  STDatasets(2)%VarNames = lwdnNames(1:lwdnNo)
+  STDatasets(3)%VarNames = swdnNames(1:swdnNo)
+  STDatasets(4)%VarNames = presNames(1:presNo)
+  STDatasets(5)%VarNames = qairNames(1:qairNo)
+  STDatasets(6)%VarNames = TmaxNames(1:TmaxNo)
+  STDatasets(7)%VarNames = TminNames(1:TminNo)
+  STDatasets(8)%VarNames = uwindNames(1:uwindNo)
+  STDatasets(9)%VarNames = vwindNames(1:vwindNo)
+  STDatasets(10)%VarNames = fdiffNames(1:fdiffNo)
+END SUBROUTINE read_variable_names
+
+!------------------------------------------------------------------------------!
+
+SUBROUTINE read_landmask(LandmaskFile, CRU)
+  !*## Purpose
+  !
+  ! Reads a landmask and allocates the data arrays in the ```CRU``` derived
+  ! type, sets up indexing arrays for reading from met forcing datasets and
+  ! global landmask variables.
+  !
+  !*## Method
+  !
+  ! Uses the NetCDF dataset at ```LandmaskFile``` to determine which cells are
+  ! computed in this run of CABLE.
+
+  ! Read the landmask file to determine which land cells we're looking at.
+  ! Define the input variables
+  CHARACTER(len=256), INTENT(IN)  :: LandmaskFile
+  TYPE(cru_type), INTENT(INOUT)   :: CRU
+
+  ! Assign an integer to store the file ID, and to record the error status
+  INTEGER       :: FileID, ErrStatus
+
+  ! And some IDs to store the latitude, longitude and mask variable IDs
+  INTEGER       :: LatID, LonID, LandID
+
+  ! Need the lengths of the axes
+  !INTEGER       :: xDimSize, yDimSize
+
+  ! And then the actual values along the axes
+  REAL, DIMENSION(:), ALLOCATABLE :: Latitudes, Longitudes
+
+  ! Need a counter and iterators when iterating through the points in the land mask
+  INTEGER       :: MaskCounter, LatIndx, LonIndx, VarIndx
+
+  INTEGER, DIMENSION(:,:), ALLOCATABLE :: LandMask
+
+  ! Now we attempt to read the netCDF file
+  ErrStatus = NF90_OPEN(TRIM(LandmaskFile), nf90_nowrite, FileID)
+  IF (ErrStatus /= NF90_NOERR) THEN
+    CALL NC_ABORT(ErrStatus, "Error reading the landmask file.")
+  END IF
```

Comment:
Why use `NC_ABORT()` and not `handle_err()`? I would settle on using only one. Was that a copy/paste code already there?

The comment is valid for the other `NC_ABORT()` calls in this file.

```suggestion
  CALL handle_err(ErrStatus, "Error reading the landmask file.")
```